src/lib/Libifl/ifl_util.c

```diff
@@ -135,40 +141,64 @@ random_srv_conn(svr_conn_t **svr_conns)
 }
 
 /**
- * @brief	Get the server instance index from the jobid, which will act as a hint
- * 			as to which server the job might possibly be located
+ * @brief	Get the server instance index from the job/resv id, which will act as a hint
+ * 		as to which server the job/reservation might possibly be located.
+ * 		This function also works even if PBS_SERVER name is not part of obj_id.
+ * 		For example, obj_id can be "123.stblr3" or "123", "R123.stblr4" or "R123" where
+ * 		stblr3/stblr4 are PBS_SERVER names.
 
- * @param[in] id - object id
+ * @param[in] obj_id - job or resv object id
+ * @param[in] obj_type - job or resv object
  *
  * @return int
  * @retval >= 0: start_ind - index where the request should be fired first
  * @retval < 0: could not find appropriate index
  */
 int
-get_job_location_hint(char *jobid)
+get_shard_obj_location_hint(char *obj_id, int obj_type)
 {
 	int nsvrs = get_num_servers();
 	char *ptr = NULL;
+	char *ptr_idx = NULL;
 	int svridx = -1;
 	char *endptr = NULL;
+	int id_len = 0;
```

Comment:
Now that I am implementing your next comment https://github.com/openpbs/openpbs/pull/2255#discussion_r582375355, I think it is better to leave it as "id_len" only. Because we are not decrementing id_len value.